docs/trie.md

```diff
@@ -0,0 +1,77 @@
+# Trie
+
+We use Merkle-Patricia Trie to store blockchain state.
+Trie is persistent, which means that insertion of new node actually leads to creation of new path to this node, and thus root of Trie after insertion will also be presented by new object.
+
+Here we describe its implementation details which are closely related to Runtime.
+
+## Main structures
+
+### Trie
+
+Base structure. Almost never called directly from `Runtime`, modifications are made using `TrieUpdate`.
```

Comment:
This misses "big picture" info about how the data is actually stored, while it may be obvious to maintaners, it's a crucial missing knowledge for new contributors. 

Let's say that we store trie in the rocksdb database, which is persistet across node restarts, and metion specific columns which we use for the storage. 

That is, the document focuses primarily on the in-code abstraction we use in the code to represent the trie, but it is somewhat more useful, in these sort of documents, do describe the underlying thing in the real world that the abstraction models. In this case, that would be a bit of persistent state in the rocksdb.